pkg/eosclient/eosbinary/eosbinary.go

```diff
@@ -736,18 +736,23 @@ func (c *Client) Write(ctx context.Context, auth eosclient.Authorization, path s
 	if err != nil {
 		return err
 	}
-	return c.writeFile(ctx, auth, path, fd.Name(), app)
+	return c.writeFile(ctx, auth, path, fd.Name(), app, disableVersioning)
 }
 
 // WriteFile writes an existing file to the mgm.
-func (c *Client) writeFile(ctx context.Context, auth eosclient.Authorization, path, source, app string) error {
+func (c *Client) writeFile(ctx context.Context, auth eosclient.Authorization, path, source, app string, disableVersioning bool) error {
 	xrdPath := fmt.Sprintf("%s//%s", c.opt.URL, path)
 	args := []string{"--nopbar", "--silent", "-f", source, xrdPath}
 
 	if auth.Token != "" {
 		args[4] += "?authz=" + auth.Token
 	} else if auth.Role.UID != "" && auth.Role.GID != "" {
-		args = append(args, fmt.Sprintf("-ODeos.ruid=%s&eos.rgid=%s&eos.app=%s", auth.Role.UID, auth.Role.GID, app))
+		queryParams := fmt.Sprintf("-ODeos.ruid=%s&eos.rgid=%s&eos.app=%s", auth.Role.UID, auth.Role.GID, app)
+
+		if disableVersioning {
```

Comment:
Why is this dependent on the auth type? We will switch from gateway-based auth to token-based, so I imagine that would fall under the previous if, and therefore there would be no version disabling possibility?